src/train/KWIC.py

```diff
@@ -0,0 +1,245 @@
+    width = 10
+
+    outstring = ' '.join(kwic[:keyindex]).rjust(width * keyindex)
+    outstring += str(kwic[keyindex]).center(len(kwic[keyindex]) + 6)
+    outstring += ' '.join(kwic[(keyindex + 1):])
+
+    return outstring
+
+def cut_to_sentence(text, keyword, keywordindex):
+    """ Cuts the sentence around a keyword out of the text
+    Arguments
+    ----------
+    text : str
+        Text out of which the sentence should be extracted
+    keyword : str
+        Keyword in the sentence of the text
+    keywordindex: int
+        Index of the keyword in the text
+    Returns
+    -------
+    Indices of of the sentence in the text and a string of the sentence
+    """
+    # Strings after wich a point does not end a sentence
+    safe = ["Ms", "Mr", "Fr", "Hr", "Dipl", "B", "M", "Sc", "Dr", "Prof",
+            "Mo", "Mon", "Di", "Tu", "Tue", "Tues", "Mi", "Wed", "Do", "Th",
+            "Thu", "Thur", "Thurs", "Fr", "Fri", "Sa", "Sat", "So", "Sun",
+            "0", "1", "2", "3", "4", "5", "6", "7", "8", "9",
+            "str"]
+
+    # Find beginning
+    rfind_results = []
+    end_ = keywordindex
+    # Special Case "."
+    while True:
+        rfind_ = text.rfind(". ", 0, end_)
+        if not rfind_ == -1:
+            no_safe = False
+            for i, s in enumerate(safe):
+                if text[0:rfind_][::-1].find(s[::-1]) == 0:
+                    end_ = rfind_ - len(s)
+                    break
+                if i == len(safe)-1:
+                    no_safe = True
+            if no_safe is True:
+                break
+        else:
+            break
+    rfind_results.append(rfind_)
+
+    rfind_results.append(max([text.rfind(sentence_ending, 0, keywordindex)
+                              for sentence_ending in ["! ", "? "]]))
+
+    rfind_result = max(rfind_results)
+    if rfind_result == -1:
+        start = 0
+    else:
+        start = rfind_result + 2
+
+    # Find ending
+    find_results = []
+    start_ = keywordindex+len(keyword)
+    # Special Case "."
+    while True:
+        find_ = text.find(". ", start_)
+        if not find_ == -1:
+            no_safe = False
+            for i, s in enumerate(safe):
+                if text[0:find_][::-1].find(s[::-1]) == 0:
+                    start_ = find_ + len(s)
+                    break
+                if i == len(safe)-1:
+                    no_safe = True
+            if no_safe is True:
+                break
+        else:
+            break
+    find_results.append(find_)
+
+    find_results.extend([text.find(sentence_ending, keywordindex+len(keyword))
+                         for sentence_ending in ["! ", "? "]])
+    find_results_bigger_neg_1 = [i for i in find_results if i >= 0]
+    if not find_results_bigger_neg_1:
+        end = len(text)
+    else:
+        end = min(find_results_bigger_neg_1) + 1
+
+    return list(range(start, end)), text[start:end]
+
+def find_nth_occurrence(text, searchstr, nth=1, startindex=0):
```

Comment:
Is this counting the characters in a row that is returned or the number of words in a row?